raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -3329,6 +3408,23 @@ private UpdateRaftVoterRequestData buildUpdateVoterRequest() {
         );
     }
 
+    private AddRaftVoterRequestData buildAddVoterRequest() {
+        return RaftUtil.addVoterRequest(
+            clusterId,
+            quorumConfig.requestTimeoutMs(),
+            quorum.localReplicaKeyOrThrow(),
+            localListeners,
+            !quorumConfig.autoJoin()
```

Comment:
Why is this the negative of auto join? Shouldn't it always be false? If KRaft send an "add voter" request, it should always be version 1 and return before committing.